sdk/js/src/sui/publish.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  fromB64,
+  JsonRpcProvider,
+  normalizeSuiAddress,
+  normalizeSuiObjectId,
+  TransactionBlock,
+} from "@mysten/sui.js";
+import { SuiBuildOutput } from "./types";
+import { getPackageId } from "./utils";
+
+export const publishCoin = async (
+  provider: JsonRpcProvider,
+  coreBridgeStateObjectId: string,
+  tokenBridgeStateObjectId: string,
+  decimals: number,
+  signerAddress: string
+) => {
+  const coreBridgePackageId = await getPackageId(
+    provider,
+    coreBridgeStateObjectId
+  );
+  const tokenBridgePackageId = await getPackageId(
+    provider,
+    tokenBridgeStateObjectId
+  );
+  const build = getCoinBuildOutput(
+    coreBridgePackageId,
+    tokenBridgePackageId,
+    decimals
+  );
+  return publishPackage(build, signerAddress);
+};
+
+export const getCoinBuildOutput = (
+  coreBridgePackageId: string,
+  tokenBridgePackageId: string,
+  decimals: number
+): SuiBuildOutput => {
+  decimals = Math.min(decimals, 8);
+  const bytecode = Buffer.from(
+    String(
+      "a11ceb0b060000000901000a020a14031e1704350405392d07669f01088502600ae502050cea02160004010b010c0205020d000002000201020003030c020001000104020700000700010001090801010c020a050600030803040202000302010702080007080100020800080303090002070801010b020209000901010608010105010b0202080008030209000504434f494e095478436f6e7465787408565f5f305f315f3111577261707065644173736574536574757004636f696e0e6372656174655f777261707065640b64756d6d795f6669656c6404696e697414707265706172655f726567697374726174696f6e0f7075626c69635f7472616e736665720673656e646572087472616e736665720a74785f636f6e746578740f76657273696f6e5f636f6e74726f6c0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000268da393248d51fbe2bd7456414adefdf7eac9ef91b32b561c178f01c906ae80e00020106010000000001090b00310" +
```

Comment:
`hexdump -C` shows me the following for the above bytecode (I put in 8 for decimals)
```
00000000  a1 1c eb 0b 06 00 00 00  09 01 00 0a 02 0a 14 03  |................|
00000010  1e 17 04 35 04 05 39 2d  07 66 9f 01 08 85 02 60  |...5..9-.f.....`|
00000020  0a e5 02 05 0c ea 02 16  00 04 01 0b 01 0c 02 05  |................|
00000030  02 0d 00 00 02 00 02 01  02 00 03 03 0c 02 00 01  |................|
00000040  00 01 04 02 07 00 00 07  00 01 00 01 09 08 01 01  |................|
00000050  0c 02 0a 05 06 00 03 08  03 04 02 02 00 03 02 01  |................|
00000060  07 02 08 00 07 08 01 00  02 08 00 08 03 03 09 00  |................|
00000070  02 07 08 01 01 0b 02 02  09 00 09 01 01 06 08 01  |................|
00000080  01 05 01 0b 02 02 08 00  08 03 02 09 00 05 04 43  |...............C|
00000090  4f 49 4e 09 54 78 43 6f  6e 74 65 78 74 08 56 5f  |OIN.TxContext.V_|
000000a0  5f 30 5f 31 5f 31 11 57  72 61 70 70 65 64 41 73  |_0_1_1.WrappedAs|
000000b0  73 65 74 53 65 74 75 70  04 63 6f 69 6e 0e 63 72  |setSetup.coin.cr|
000000c0  65 61 74 65 5f 77 72 61  70 70 65 64 0b 64 75 6d  |eate_wrapped.dum|
000000d0  6d 79 5f 66 69 65 6c 64  04 69 6e 69 74 14 70 72  |my_field.init.pr|
000000e0  65 70 61 72 65 5f 72 65  67 69 73 74 72 61 74 69  |epare_registrati|
000000f0  6f 6e 0f 70 75 62 6c 69  63 5f 74 72 61 6e 73 66  |on.public_transf|
00000100  65 72 06 73 65 6e 64 65  72 08 74 72 61 6e 73 66  |er.sender.transf|
00000110  65 72 0a 74 78 5f 63 6f  6e 74 65 78 74 0f 76 65  |er.tx_context.ve|
00000120  72 73 69 6f 6e 5f 63 6f  6e 74 72 6f 6c 00 00 00  |rsion_control...|
00000130  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
00000160  00 00 00 00 00 00 00 00  00 00 00 00 02 68 da 39  |.............h.9|
00000170  32 48 d5 1f be 2b d7 45  64 14 ad ef df 7e ac 9e  |2H...+.Ed....~..|
00000180  f9 1b 32 b5 61 c1 78 f0  1c 90 6a e8 0e 00 02 01  |..2.a.x...j.....|
00000190  06 01 00 00 00 00 01 09  0b 00 31 08 0a 01 38 00  |..........1...8.|
000001a0  0b 01 2e 11 02 38 01 02  00                       |.....8...|
```
notice how the version identifier is `V_0_1_1` - do we not want to parameterise by that too?

Reply:
It would actually need to be the fully qualified type sine a new one gets introduced every package version, but good thought, it's on the radar 🙂

Reply:
yeah that's what I meant, of course the new versions won't be available from the old package, but the version type should ideally be a parameter

Reply:
Agreed. We’re on it

Reply:
Unfortunately, parametrizing by version struct seems not possible because it alters other parts of the bytecode as well :(